sass_builder_test.dart

```diff
@@ -0,0 +1,207 @@
+import 'package:build/build.dart';
+import 'package:build_barback/build_barback.dart';
+import 'package:build_test/build_test.dart';
+import 'package:sass_builder/sass_builder.dart';
+import 'package:test/test.dart';
+
+void main() {
+  // These tests only verify which assets are read and written by the
+  // SassBuilder. This is because the inputs are being parsed to find what files
+  // they import.
+  //
+  // These tests do not verify any output as that is determined by the sass
+  // implementation.
+  group('File import parsing tests', () {
+    SassBuilder builder;
+    InMemoryAssetWriter writer;
+    InMemoryAssetReader reader;
+
+    setUp(() {
+      builder = new SassBuilder();
+      reader = new InMemoryAssetReader();
+      writer = new InMemoryAssetWriter();
+    });
+
+    test('no imports, one read and one write', () async {
+      var primary = makeAssetId('a|lib/styles.scss');
+      var inputs = {
+        primary: '/* no imports */',
+      };
+
+      reader.cacheStringAsset(primary, inputs[primary]);
+
+      await runBuilder(builder, inputs.keys, reader, writer, const BarbackResolvers());
+
+      expect(writer.assets.length, equals(1));
+      expect(writer.assets, contains(primary.changeExtension('.css')));
+
+      expect(reader.assets.length, equals(1));
+      expect(reader.assetsRead, contains(primary));
+     });
+
+    test('one relative partial import', () async {
+      var primary = makeAssetId('a|lib/styles.scss');
+      var import = makeAssetId('a|lib/_more_styles.scss');
+      var inputs = {
+        primary: '''@import '_more_styles.scss';''',
+        import: '''/* no imports */''',
+      };
+
+      reader.cacheStringAsset(primary, inputs[primary]);
+      reader.cacheStringAsset(import, inputs[import]);
+
+      await runBuilder(builder, inputs.keys, reader, writer, const BarbackResolvers());
+
+      expect(writer.assets.length, equals(1));
+      expect(writer.assets, contains(primary.changeExtension('.css')));
+
+      expect(reader.assets.length, equals(2));
+      expect(reader.assetsRead, contains(primary));
```

Comment:
nit: `allOf(contains(primary), contains(import))` or `unorderedEquals([primary, import])` read a bit better to me (same below)